### DISCUSSION TITLE
Is using classes for backend/server a good idea in Next.js?

### DISCUSSION THREAD

**Discussion by Firgrep** [vercel/next.js#72028] [Help] [answered]
### Summary

Myself and others in my team like to pack various backend operations into classes and methods instead of functions. Say I have a bunch of stripe or db functions that I put into a class with static methods that I use around my app as a singleton service. 

I realize that big classes may not be ideal when it comes to tree-shaking and bundlers, as the whole thing gets shipped if a component uses only one method (correct me if I'm wrong, please). 

But what is the case when it comes to code running on the server, say route handlers or, particularly, server components when it comes to Next.js? Are they bundled the same way as client components? Should I go for independent functions or is it okay to use big classes when it comes to the server, API routes or server components? Thank you.

**Comment by aniruddhaadak80** [comment, marked as answer]
Hey @Firgrep ,

Using classes for backend logic in Next.js server components and API routes is generally fine:

1. **Tree-shaking**: Not an issue on the server side, as unused code won’t be bundled for client components.
2. **Server-only Bundling**: Next.js effectively separates server-side and client-side code, so large classes on the server won’t increase client bundle size.
3. **Maintainability**: Classes with static methods are helpful for organizing related operations, like database or API calls.

For backend/server logic, classes are fine if kept focused and modular.

Thank you.

**Reply by Firgrep** [reply]
Hey @aniruddhaadak80 thank you for your reply, it's exactly what I needed to know. Cheers!

**Reply by aniruddhaadak80** [reply]
I am very happy @Firgrep , that I could help you .🎉
